drivers/dma/dma_stm32.c

```diff
@@ -493,9 +495,11 @@ DMA_STM32_EXPORT_API int dma_stm32_reload(const struct device *dev, uint32_t id,
 		return -EINVAL;
 	}
 
-	stm32_dma_disable_stream(dma, id);
 	stream = &config->streams[id];
 
+	while (stm32_dma_disable_stream(dma, id) != 0) {
```

Comment:
Here, too.

Reply:
Now used the available `dma_stm32_disable_stream` function, that has a finite loop, so now it should not hang forever if something goes wrong.